### PR TITLE
Am/chore/yippee ki yay

### PR DESCRIPTION
**Pull request by IceTDrinker**
### PR content/description

Migrate concrete-csprng from concrete-core repo
Add dieharder randomness test with a workflow

### Check-list:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by IceTDrinker** [comment]
@slab-ci csprng_randomness_testing

**Comment by IceTDrinker** [comment]
@slab-ci csprng_randomness_testing

**Comment by IceTDrinker** [comment]
@slab-ci csprng_randomness_testing

**Comment by IceTDrinker** [comment]
@slab-ci csprng_randomness_testing

**Comment by IceTDrinker** [comment]
@slab-ci csprng_randomness_testing

**Comment by IceTDrinker** [comment]
@slab-ci csprng_randomness_testing

**Comment by IceTDrinker** [comment]
@slab-ci csprng_randomness_testing

**Comment by IceTDrinker** [comment]
from placeholder workflow run : https://github.com/zama-ai/tfhe-rs/actions/runs/6035443311/job/16375793784

File: Makefile [hunk 1]
```diff
@@ -80,6 +79,15 @@ install_node:
 	$(SHELL) -i -c 'nvm install node' || \
 	( echo "Unable to install node, unknown error." && exit 1 )
 
+.PHONY: install_dieharder # Install dieharder for apt distributions or macOS
+install_dieharder:
+	@dieharder -h > /dev/null 2>&1 || \
+	if [[ "$(OS)" == "Linux" ]]; then \
+		sudo apt update && sudo apt install -y dieharder; \
+	elif [[ "$(OS)" == "Darwin" ]]; then\
+		echo "lol"; \
```
Comment (IceTDrinker) [inline Makefile:88]: ah

File: Makefile [hunk 2]
```diff
@@ -3,8 +3,7 @@ OS:=$(shell uname)
 RS_CHECK_TOOLCHAIN:=$(shell cat toolchain.txt | tr -d '\n')
 CARGO_RS_CHECK_TOOLCHAIN:=+$(RS_CHECK_TOOLCHAIN)
 TARGET_ARCH_FEATURE:=$(shell ./scripts/get_arch_feature.sh)
-RS_BUILD_TOOLCHAIN:=$(shell \
-	( (echo $(TARGET_ARCH_FEATURE) | grep -q x86) && echo stable) || echo $(RS_CHECK_TOOLCHAIN))
+RS_BUILD_TOOLCHAIN:=stable
```
Comment (soonum) [inline Makefile:6]: Could you explain the rationale behind this simplification ?

File: README.md [hunk 3]
```diff
@@ -47,7 +47,7 @@ tfhe = { version = "*", features = ["boolean", "shortint", "integer", "x86_64-un
 ```toml
 tfhe = { version = "*", features = ["boolean", "shortint", "integer", "aarch64-unix"] }
 ```
-Note: users with ARM devices must use `TFHE-rs` by compiling using the `nightly` toolchain.
+Note: users with ARM devices must compile `TFHE-rs` using a stable toolchain with version >= 1.72.
```
Comment (soonum) [inline README.md:50]: I guess this is the explanation for my earlier comment...

**Review by soonum — COMMENTED** [review]
From CI point of view everything seems right. I'll @tmontaigu perform the review on the implementation.

File: Makefile (hunk 1 above)
Comment (soonum) [reply Makefile:88]: I missed that :frowning:

File: Makefile (hunk 2 above)
Comment (IceTDrinker) [reply Makefile:6]: concrete-csprng now does not require a nighlty toolchain for aarch64, so everybody should use the stable toolchain !

File: README.md (hunk 3 above)
Comment (IceTDrinker) [reply README.md:50]: indeed

**Review by tmontaigu — APPROVED** [review]

**Comment by IceTDrinker** [comment]
need a rebase for the Makefile 😭 @tmontaigu I'll need another approve so that tests are triggered

**Review by tmontaigu — APPROVED** [review]

**Comment by IceTDrinker** [comment]
forgot parallel feature for csprng tests -_-

File: Makefile [hunk 4]
```diff
@@ -342,6 +362,11 @@ test_kreyvium: install_rs_build_toolchain
 		kreyvium --features=$(TARGET_ARCH_FEATURE),boolean,shortint,integer \
 		-- --test-threads=1
 
+.PHONY: test_concrete_csprng # Run concrete-csprng tests
+test_concrete_csprng:
+	RUSTFLAGS="$(RUSTFLAGS)" cargo $(CARGO_RS_BUILD_TOOLCHAIN) test --profile $(CARGO_PROFILE) \
+		--features=$(TARGET_ARCH_FEATURE) -p concrete-csprng
```
Comment (IceTDrinker) [inline Makefile:368]: missing parallel feature

Comment (IceTDrinker) [reply Makefile:368]: ah no all good, it's already set via the target arch feature

**Pull request merged**